api/tests/unit/domain/usecases/deactivate-admin-member_test.js

```diff
@@ -4,17 +4,38 @@ const deactivateAdminMember = require('../../../../lib/domain/usecases/deactivat
 describe('Unit | UseCase | deactivate-admin-member', function () {
   it('should deactivate the given admin member', async function () {
     // given
-    const adminMemberRepository = { deactivate: sinon.stub() };
+    const adminMemberRepository = { deactivate: sinon.stub(), getById: sinon.stub().resolves({ userId: 2 }) };
+    const refreshTokenService = { revokeRefreshTokensForUserId: sinon.stub() };
 
     adminMemberRepository.deactivate.withArgs({ id: 7 }).resolves(undefined);
 
     // when
     const adminMember = await deactivateAdminMember({
       id: 7,
       adminMemberRepository,
+      refreshTokenService: refreshTokenService,
     });
 
     // then
     expect(adminMember).to.be.undefined;
+    expect(refreshTokenService.revokeRefreshTokensForUserId).to.have.been.calledWithExactly({ userId: 2 });
+  });
+
+  it("should revoke all user's refresh tokens", async function () {
+    // given
+    const adminMemberRepository = { deactivate: sinon.stub(), getById: sinon.stub().resolves({ userId: 2 }) };
+    const refreshTokenService = { revokeRefreshTokensForUserId: sinon.stub() };
+
+    adminMemberRepository.deactivate.withArgs({ id: 7 }).resolves(undefined);
+
+    // when
+    await deactivateAdminMember({
+      id: 7,
+      adminMemberRepository,
+      refreshTokenService: refreshTokenService,
+    });
+
+    // then
+    expect(refreshTokenService.revokeRefreshTokensForUserId).to.have.been.calledWithExactly({ userId: 2 });
```

Comment:
**Remarque** : On expect à peu de chose près la même chose dans ces deux tests, que la fonction est bien appelé avec les paramètres attendus.

**Proposition** : Ne faire qu'un test qui couvre les 2 premiers appels avec un with args dans le given (fait pour le deactivate mais pas le getById).


```
// given
    const adminMemberRepository = { deactivate: sinon.stub(), getById: sinon.stub() };
    const refreshTokenService = { revokeRefreshTokensForUserId: sinon.stub() };

    adminMemberRepository.deactivate.withArgs({ id: 7 }).resolves(undefined);
    adminMemberRepository.getById.withArgs(7).resolves({ userId: 2 });

...

// then
    expect(adminMember).to.be.undefined;
    expect(refreshTokenService.revokeRefreshTokensForUserId).to.have.been.calledWithExactly({ userId: 2 });
```

Reply:
Je rajoute le withArgs.
Par contre, dans plein d'autres cas et même en mob, on a plutôt l'habitude de faire des tests pour chaque comportement. Là, tu veux que je fasse 2 assertions dans un seul test ? Comme ça 


  it("should deactivate the given admin member and revoke all user's refresh tokens", async function () {
    // given
    const adminMemberRepository = { deactivate: sinon.stub(), getById: sinon.stub() };
    adminMemberRepository.deactivate.withArgs({ id: 7 }).resolves(undefined);
    adminMemberRepository.getById.withArgs(7).resolves({ userId: 2 });

    const refreshTokenService = { revokeRefreshTokensForUserId: sinon.stub() };

    // when
    const adminMember = await deactivateAdminMember({
      id: 7,
      adminMemberRepository,
      refreshTokenService: refreshTokenService,
    });

    // then
    expect(adminMember).to.be.undefined;
    expect(refreshTokenService.revokeRefreshTokensForUserId).to.have.been.calledWithExactly({ userId: 2 });
  });

Reply:
Si je lis le code existant c'est déjà ce que tu fais dans les premier test `should deactivate the given admin member`  tu expect les deux, d'où ma proposition d'en faire un test. Le deuxième fait doublon.

Sauf si tu ne voulais pas expect le revokeRefresh dans le premier test mais le deactivate :)

Reply:
C'est plus clair, merci